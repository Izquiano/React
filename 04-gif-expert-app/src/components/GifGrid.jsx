import React from 'react'
import { useFechGifs } from '../hooks/useFetchGifs'

import { GifGridItem } from './GifGridItem'

export const GifGrid = ( {category} ) => {

  

  const { data: images, loading } = useFechGifs( category )

  // useEffect( ()=> {
  //   getGifs( category )
  //   .then( imgs => setImages( imgs ))
  // }, [ category ] )

  

  return (
    <>
    <h3>{ category }</h3>
    {loading && <p>Loading...</p> }
  

    <div className="card-grid">
           
       {
         images.map( el => {
           return (
            <GifGridItem 
              key={ el.id } 
              { ...el }
            />
           )
         })
       }
       
    
    </div>
    </>
  )
}
