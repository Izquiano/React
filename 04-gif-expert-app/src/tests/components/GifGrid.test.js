import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFechGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas sobre el componente <GifGrid />", () => {
  const category = "One Punch";

  test("Debe mostrar el componente correctamente", () => {
    useFechGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostar items cuando se cargan imÃ¡genes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://locakhost/culaquier/cosa.jpg",
        title: "cualquier cosa",
      },
      {
        id: "BFS",
        url: "http://locakhost/culaquier/otracosa.jpg",
        title: "cualquier otra cosa",
      },
    ];

    useFechGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('p').exists() ).toBe( false )

    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )


  });
});
