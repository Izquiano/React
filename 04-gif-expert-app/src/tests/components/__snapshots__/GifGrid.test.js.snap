// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> Debe de mostar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="http://locakhost/culaquier/cosa.jpg"
    />
    <GifGridItem
      id="BFS"
      key="BFS"
      title="cualquier otra cosa"
      url="http://locakhost/culaquier/otracosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas sobre el componente <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
